packages/vertexai/src/requests/response-helpers.ts

```diff
@@ -196,3 +199,52 @@ export function formatBlockErrorMessage(
   }
   return message;
 }
+
+/**
+ * Convert a generic successful fetch response body to an Imagen response object
+ * that can be returned to the user. This converts the REST APIs response format to our
+ * APIs representation of a response.
+ *
+ * @internal
+ */
+
+export async function handlePredictResponse<
+  T extends ImagenInlineImage | ImagenGCSImage
+>(response: Response): Promise<{ images: T[]; filteredReason?: string }> {
+  const responseJson: ImagenResponseInternal = await response.json();
+
+  const images: T[] = [];
+  let filteredReason: string | undefined = undefined;
+
+  if (!responseJson.predictions || responseJson.predictions?.length === 0) {
+    throw new VertexAIError(
+      VertexAIErrorCode.ERROR,
+      "Predictions array is undefined or empty in response. Was 'includeRaiReason' enabled in the request?"
```

Comment:
The user can't set this property, right? Is this error useful to them? Or is it just for them to pass along to us? Also isn't this a `VertexAIErrorCode.RESPONSE_ERROR`? I mean I guess it's because the error could have been made in the request, but isn't that the case for a lot of response errors? I thought we just use it to say that an error came back in the response, as opposed to us having blocked it before the request was sent.

Reply:
You're right- the user can't set this property, so this error isn't useful to them. This case should never happen, since it would mean we got a malformed response from the backend. I think we should still throw an error if for some reason this unexpectedly does happen. I've updated this to be a `RESPONSE_ERROR`, and the error message to indicate that it's an unexpected error, and that the user should report an issue to us.

Reply:
sg!